apisix/discovery/consul/init.lua

```diff
@@ -532,7 +541,8 @@ function _M.connect(premature, consul_server, retry_delay)
                         core.table.insert(nodes, {
                             host = svc_address,
                             port = tonumber(svc_port),
-                            weight = default_weight,
+                            weight = metadata and tonumber(metadata.weight) or default_weight,
```

Comment:
Should we get the weight from the metadata? Is this a common practice? Should some comments be added?

Reply:
weight has been deleted from this PR and will submit in another, fixed